### PR TITLE
VimeoNetworking clean up

### PR DESCRIPTION
**Pull request by rpassis**
#### Ticket

[VIM-7216](https://vimean.atlassian.net/browse/VIM-7216)

#### Pull Request Checklist

- [X] Resolved any merge conflicts
- [X] No build errors or warnings are introduced
- [X] New files are written in Swift
- [X] New classes contain license headers
- [X] New classes have Documentation
- [X] New public methods have Documentation

#### Issue Summary

While doing some prep research on VimeoNetworking I came across old/unused code that could be removed. The purpose of this ticket is to account for the clean up work.

Also see [related PR](https://github.vimeows.com/MobileApps/VimeoKit/pull/468/) on VimeoKit

#### Implementation Summary

- Most of the changes are for removing unused code
- Added nullability annotations to VIMModelObject
- Refactored Mappable conformance to avoid duplicated implementation of validateModel()

#### How to Test

- Build and run, make sure all unit tests pass
- Update the main Vimeo app to point to this branch on both `VimeoNetworking` and `VimeoKit`, build and run and ensure all tests pass

### CODE REVIEW

**Comment by cameo-github** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">SwiftLint not installed, download from https://github.com/realm/SwiftLint</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 117 tests, with 0 failures (0 unexpected) in 1.422 (1.757) seconds</td>
    </tr>
  </tbody>
</table>

See build details on [CircleCI](https://circleci.com/gh/vimeo/VimeoNetworking/1206)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: Sources/Shared/Mappable.swift [hunk 1]
```diff
@@ -62,60 +59,26 @@ extension VIMModelObject: MappableResponse {
...
  */
-extension Array: MappableResponse {
-    // The default implementation for all arrays will return no mapping class or model key path
-    // Only if Element itself is a VIMModelObject will the values be returned
-    // TODO: This is because we can't restrict the generic type if we're
-    // extending a type with generics to conform to a protocol [RH]
-    
+extension Array: MappableResponse where Element: VIMModelObject {
+
     public static var mappingClass: AnyClass? {
```
Comment (Jumhyn) [inline Sources/Shared/Mappable.swift]: Do we need to continue to support the optionality here, or could `MappableResponse.mappingClass` just have type `AnyClass`?

**Review by Jumhyn — APPROVED** [review]
This looks good. One small question/suggestion but I'm ambivalent as to whether it gets picked up in this PR, even if it is a possible improvement. Builds and tests A-OK.

File: Sources/Shared/Mappable.swift (hunk 1 above)
Comment (rpassis) [reply Sources/Shared/Mappable.swift]: @Jumhyn that was a great catch, updated and PTAL!

**Review by Jumhyn — APPROVED** [review]
Awesome, LGTM.

**Pull request merged**